### DISCUSSION TITLE
How to dynamically select database connections based on requests?

### DISCUSSION THREAD

**Discussion by lanrenbulan** [symfony/symfony#44812] [Q&A] [answered]
Hi,

My application is SaaS. There are about 1000 tenants in our old system. How to dynamically create database connections according to sessions in symfony?

For instance:
Save the tenant ID in seesion and read the database configuration information in redis according to the tenant ID

What can be done? For example, self defining a connection?

doctrine.dbal.wrapper_class

**Comment by ro0NL** [comment, marked as answer]
See eg. https://karoldabrowski.com/blog/dynamic-database-connection-based-on-request-symfony-and-doctrine/
